internal/services/eventhub/eventhub_namespace_customer_managed_key_resource.go

```diff
@@ -174,14 +207,29 @@ func resourceEventHubNamespaceCustomerManagedKeyRead(d *pluginsdk.ResourceData,
 
 		d.Set("key_vault_key_ids", keyVaultKeyIds)
 		d.Set("infrastructure_encryption_enabled", props.Encryption.RequireInfrastructureEncryption)
+
+		if props.Encryption.KeyVaultProperties != nil {
+			// we can only have a single user managed id for N number of keys, azure portal only allows setting a single one and then applies it to each key
+			for _, item := range *props.Encryption.KeyVaultProperties {
+				if item.Identity != nil {
+					userAssignedId, err := commonids.ParseUserAssignedIdentityIDInsensitively(*item.Identity.UserAssignedIdentity)
```

Comment:
and UserAssignedIdentity here?

```suggestion
					if (item.Identity.UserAssignedIdentity != nil) {
					userAssignedId, err := commonids.ParseUserAssignedIdentityIDInsensitively(*item.Identity.UserAssignedIdentity)
```